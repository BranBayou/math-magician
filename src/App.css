.container {
  max-width: 920px;
  height: 100%;
  margin: 0 auto;
}
