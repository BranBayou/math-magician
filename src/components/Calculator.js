import './Calculator.css';

function Calculator() {
  return (
    <div className="cal-con">
      <input className="btn" id="formula" type="text" placeholder="0" />
      <input className="btn" type="button" value="AC" />
      <input className="btn" type="button" value="+/-" />
      <input className="btn" type="button" value="%" />
      <input className="btn math-opr" type="button" value="/" />
      <input className="btn" type="button" value="7" />
      <input className="btn" type="button" value="8" />
      <input className="btn" type="button" value="9" />
      <input className="btn math-opr" type="button" value="x" />
      <input className="btn" type="button" value="4" />
      <input className="btn" type="button" value="5" />
      <input className="btn" type="button" value="6" />
      <input className="btn math-opr" type="button" value="-" />
      <input className="btn" type="button" value="1" />
      <input className="btn" type="button" value="2" />
      <input className="btn" type="button" value="3" />
      <input className="btn math-opr" type="button" value="+" />
      <input className="btn" id="zero" type="button" value="0" />
      <input className="btn" type="button" value="." />
      <input className="btn math-opr" type="button" value="=" />
    </div>
  );
}

export default Calculator;
