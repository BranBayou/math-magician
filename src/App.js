import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Calculator from './components/Calculator';
import GetQuotes from './components/GetQuotes';
import Nav from './components/Nav';
import Home from './components/Home';

function App() {
  return (
    <Router>
      <div className="main-container">
        <Nav />
        <div className="container">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/Calculator" element={<Calculator />} />
            <Route path="/GetQuotes" element={<GetQuotes />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
}

export default App;
