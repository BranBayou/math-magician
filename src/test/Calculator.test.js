import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';
import App from '../App';

describe('test Calculator', () => {
  it('Should render correctly', () => {
    const { getByDisplayValue } = render(<Calculator />);
    expect(getByDisplayValue('AC')).toBeInTheDocument();
    expect(getByDisplayValue('+/-')).toBeInTheDocument();
    expect(getByDisplayValue('%')).toBeInTheDocument();
    expect(getByDisplayValue('รท')).toBeInTheDocument();
    expect(getByDisplayValue('x')).toBeInTheDocument();
    expect(getByDisplayValue('-')).toBeInTheDocument();
    expect(getByDisplayValue('+')).toBeInTheDocument();
    expect(getByDisplayValue('.')).toBeInTheDocument();
    expect(getByDisplayValue('=')).toBeInTheDocument();
  });
});

test('Snapshot test', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
