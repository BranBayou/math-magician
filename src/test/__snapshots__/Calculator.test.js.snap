// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div>
  <div
    class="main-container"
  >
    <div
      class="nav-bar"
    >
      <h1>
        Math-Magicians
      </h1>
      <div
        class="links"
      >
        <li
          class="break"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="break"
        >
          <a
            href="/Calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/GetQuotes"
          >
            Quote
          </a>
        </li>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="home-container"
      >
        <h2>
          Welcome to our page!
        </h2>
        <p>
          This web calculator app is a web-based application that allows users to perform mathematical calculations. It is a simple and easy-to-use tool that can be used for a variety of tasks, such as basic arithmetic, scientific calculations, and financial analysis.
        </p>
        <p>
          Web calculator apps are typically accessed through a web browser and do not require any software to be installed on the user computer. This makes them a convenient and accessible option for users who need to perform calculations on a variety of devices, such as laptops, tablets, and smartphones.
        </p>
      </div>
    </div>
  </div>
</div>
`;
